php/class-customize-concurrency.php

```diff
@@ -14,4 +14,299 @@
+
+		?>
+		<script type="text/html" id="tmpl-customize-post-section-notifications">
+			<ul>
+				<# _.each( data.notifications, function( notification ) { #>
+					<li class="notice notice-{{ notification.type || 'info' }} {{ data.altNotice ? 'notice-alt' : '' }}" data-code="{{ notification.code }}" data-type="{{ notification.type }}">
+						<# if ( /setting_update_conflict/.test( notification.code ) ) { #>
+							<button class="button override-post-conflict" type="button"><?php esc_html_e( 'Override', 'customize-posts' ); ?></button>
+							<# } #>
+								{{ notification.message || notification.code }}
+					</li>
+					<# } ); #>
+			</ul>
+		</script>
+		<?php
+	}
+
+
+	/**
+	 * Prevent sanitize_title() on the post_name.
+	 *
+	 * @see wp_insert_post()
+	 * @filter wp_insert_post_data
+	 *
+	 * @param array $data    An array of slashed post data.
+	 * @param array $postarr An array of sanitized, but otherwise unmodified post data.
+	 * @return array
+	 */
+	public function preserve_inserted_post_name( $data, $postarr ) {
+		if ( self::POST_TYPE === $data['post_type'] && isset( $postarr['post_name'] ) ) {
+			$data['post_name'] = wp_slash( $postarr['post_name'] );
+		}
+		return $data;
+	}
+
+	public function customize_save ( \WP_Customize_Manager $wp_customize ) {
+
+		$post_values = $wp_customize->unsanitized_post_values();
+		$saved_settings = $this->get_saved_settings( array_keys( $post_values ) );
+		$timestamps = isset( $_POST['concurrency_timestamps'] ) ? (array) json_decode( wp_unslash( $_POST['concurrency_timestamps'] ) ) : array();
+
+		$invalidities = array();
+
+		foreach ( $saved_settings as $setting_id => $saved_setting ) {
+			$is_conflicted = (
+				isset( $saved_setting[ 'timestamp' ], $saved_setting[ 'value' ] )
+				&&
+				$saved_setting['timestamp'] > $timestamps[ $setting_id ]
+				&&
+				$saved_setting['value'] !== $post_values[ $setting_id ]
+			);
+			if ( $is_conflicted ) {
+				$user = get_user_by( 'ID', (int) $saved_setting['author'] );
+				$message = sprintf( 'Value from %s: %s', $user->user_nicename, $saved_setting['value'] );
+				$invalidities[ $setting_id ] = new \WP_Error( 'concurrency_conflict', $message, array( 'their_value' => $saved_setting['value'] ) );
+			}
+		}
+
+		if ( ! empty( $invalidities ) ) {
+			$exported_setting_validities = array_map( array( $wp_customize, 'prepare_setting_validity_for_js' ), $invalidities );
+			$invalid_setting_count = count( $exported_setting_validities );
+			$response = array(
+				'setting_validities' => $exported_setting_validities,
+				'message' => sprintf(
+					_n( 'There is %s conflicting setting.', 'There are %s conflicting settings.', $invalid_setting_count ),
+					number_format_i18n( $invalid_setting_count )
+				),
+			);
+
+			/** This filter is documented in wp-includes/class-wp-customize-manager.php */
+			$response = apply_filters( 'customize_save_response', $response, $this );
+			wp_send_json_error( $response );
```

Comment:
@lgedeon here's the Trac ticket for adding guarantees that `customize_validate_{$setting->id}` will get applied: https://core.trac.wordpress.org/ticket/37638

I think it's fine to assume that it will get applied in the vast majority of the cases. If we find any examples to the contrary, we should patch any `validate` methods to always also call `parent::validate( $value )`.